mock-data-provider-plugin/src/main/java/io/mosip/certify/mockdataprovider/integration/service/MockDataProviderPlugin.java

```diff
@@ -0,0 +1,139 @@
+
+    public static final String UTC_DATETIME_PATTERN = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
+
+    public static final String CERTIFY_SERVICE_APP_ID = "CERTIFY_SERVICE";
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @Autowired
+    private KeyStore keyStore;
+
+    @Autowired
+    private KeymanagerDBHelper dbHelper;
+
+    @Autowired
+    private MockTransactionHelper mockTransactionHelper;
+
+    @Value("${mosip.certify.mock.authenticator.get-identity-url}")
+    private String getIdentityUrl;
+
+    @Value("${mosip.certify.cache.security.secretkey.reference-id}")
+    private String cacheSecretKeyRefId;
+
+    @Value("${mosip.certify.cache.security.algorithm-name}")
+    private String aesECBTransformation;
+
+    @Value("${mosip.certify.cache.secure.individual-id}")
+    private boolean secureIndividualId;
+
+    @Value("${mosip.certify.cache.store.individual-id}")
+    private boolean storeIndividualId;
+
+    @Override
+    public Map<String, Object> fetchData(Map<String, Object> identityDetails) throws DataProviderExchangeException {
+        try {
+            OIDCTransaction transaction = mockTransactionHelper.getUserInfoTransaction(identityDetails.get(ACCESS_TOKEN_HASH).toString());
+            String individualId = getIndividualId(transaction);
+            if (individualId != null) {
+                Map<String, Object> res = restTemplate.getForObject(
+                        getIdentityUrl + "/" + individualId,
+                        HashMap.class);
+                res = (Map<String, Object>) res.get("response");
+                Map<String, Object> ret = new HashMap<>();
+                ret.put("vcVer", "VC-V1");
+                ret.put("id", getIdentityUrl + "/" + individualId);
+                ret.put("UIN", individualId);
+                ret.put("fullName", res.get("fullName"));
+                ret.put("gender", res.get("gender"));
+                ret.put("dateOfBirth", res.get("dateOfBirth"));
+                ret.put("email", res.get("email"));
+                ret.put("phone", res.get("phone"));
+                ret.put("addressLine1", res.get("streetAddress"));
+                ret.put("province", res.get("locality"));
+                ret.put("region", res.get("region"));
+                ret.put("postalCode", res.get("postalCode"));
+                ret.put("face", res.get("encodedPhoto"));
+                ret.put("issuer", getIdentityUrl + "/" + individualId);
+                return ret;
+            }
+        } catch (Exception e) {
+            log.error("Failed to fetch json data for from data provider plugin", e);
+        }
+
+        throw new DataProviderExchangeException("INVALID_ACCESS_TOKEN");
+    }
+
+    protected String getIndividualId(OIDCTransaction transaction) {
+        if (!storeIndividualId)
+            return null;
+        return secureIndividualId ? decryptIndividualId(transaction.getIndividualId()) : transaction.getIndividualId();
```

Comment:
I feel this code has some minor smells, I feel we should make the same change in the older Mock VCI plugin as well. Since we are a bit close to the release, I feel it'll be okay if we do this in the next refactor as well.

Wdyt @jainhitesh9998 @Piyush7034 ?

 
```suggestion
        if (!storeIndividualId) // likely this value should also be changed to something like `isIndivdualIDEncrypted`
            return transaction.getIndividualId();
        return decryptIndividualId(transaction.getIndividualId());
```